lib/screens/settings/database_list_section.dart

```diff
@@ -57,9 +64,39 @@ class DatabaseSection extends StatelessWidget {
               title: Text('Delete all Task Lists'),
               subtitle: Text(listEntries.toString()),
             ),
+            ListTile(
+              tileColor: Colors.white,
+              onTap: () => handleDatabaseDeletion(
+                context,
+                databaseService,
+              ),
+              leading: Icon(Icons.delete_forever),
+              title: Text('Reset database'),
+              subtitle: Text(databaseService.database?.path ?? ''),
+            ),
           ],
         );
       },
     );
   }
+
+  void handleDatabaseDeletion(
+    BuildContext context,
+    DatabaseService databaseService,
+  ) async {
+    final sharedPreferences =
+        Provider.of<SharedPreferences>(context, listen: false);
+    final syncService = Provider.of<ChangeCallbackNotifier<SyncService>>(
+      context,
+      listen: false,
+    ).callbackProvider;
+    final confirmed =
+        await YesNoDialog.show(context, title: 'Delete all data?') ?? false;
+    if (confirmed) {
+      await databaseService.delete();
+      await sharedPreferences.clear();
+      await syncService.clearJob();
+      DependenciesProvider.rebuild(context);
```

Comment:
Either ask for device name after setting reset or keep the old device name. Currently the device name just becomes null.

Reply:
In that case it's ok because it wipes the whole database and guides the user afterwards to the setup screens again. You probably mean the settings reset. That's in the `settings_screen`. (And yeah, you are right, I also noticed this issue when I showed you the reset.)